PLUGINS.md

```diff
@@ -1,54 +1,153 @@
 PLUGINS
 =========
 
-NativeScript plugin is a npm package that consist of the following:
+Starting with NativeScript CLI 1.1.0, you can develop or use custom plugins in your NativeScript projects.
 
-* Metadata about the package itself: name, version, supported runtime versions, etc.
-* CommonJS module(one or many) that expose the native API via a single JavaScript API.
-* Native libraries and resources.
+* [What Are NativeScript Plugins](#what-are-nativescript-plugins)
+* [Create a Plugin](#create-a-plugin)
+  * [Directory Structure](#directory-structure)
+  * [Package.json Specification](#package-json-specification)
+* [Install a Plugin](#install-a-plugin)
+  * [Valid Plugin Sources](#valid-plugin-sources)
+  * [Installation Specifics](#installation-specifics)
+  * [Manual Steps After Installation](#manual-steps-after-installation)
+* [Use a Plugin](#use-a-plugin)
+* [Remove a Plugin](#remove-a-plugin)
+  * [Removal Specifics](#removal-specifics)
+  * [Manual Steps After Removal](#manual-steps-after-removal)
+
+## What Are NativeScript Plugins
+
+A NativeScript plugin is any npm package, published or not, that exposes a native API via JavaScript and consists of the following elements.
+
+* A `package.json` file which contains the following metadata: name, version, supported runtime versions, dependencies and others. For more information, see the [Package.json Specification](#package-json-specification) section.
+* One or more CommonJS modules that expose a native API via a single JavaScript API. For more information about Common JS modules, see the [CommonJS Wiki](http://wiki.commonjs.org/wiki/CommonJS).
+* `AndroidManifest.xml` and `Info.plist` which describe the permissions and features required or used by your app for Android and iOS, respectively.
+
+The plugin must have the directory structure, described in the [Directory Structure](#directory-structure) section.
+
+## Create a Plugin
+
+If the NativeScript framework does not expose a native API that you need, you can develop a custom plugin which exposes the required functionality. When you develop a custom plugin, keep in mind the following requirements.
+
+* The plugin must be a valid npm package.
+* The plugin must expose a built-in native API. Currently, you cannot expose native APIs available via custom native libraries.
+* The plugin must be written in JavaScript and must comply with the CommonJS specification.
```

Comment:
This is not a requirement. The plugin might be written in TypeScript and the generated code be included in the plugin package.